src/main/java/org/opensearch/ad/transport/ADCancelTaskTransportAction.java

```diff
@@ -35,25 +37,32 @@ public class ADCancelTaskTransportAction extends TransportAction<ADCancelTaskReq
     private final Logger logger = LogManager.getLogger(ADCancelTaskTransportAction.class);
     private ADTaskManager adTaskManager;
     private SDKClusterService clusterService;
+    private ExtensionsRunner extensionsRunner;
 
     @Inject
     public ADCancelTaskTransportAction(
         TaskManager taskManager,
         ActionFilters actionFilters,
         ADTaskManager adTaskManager,
-        SDKClusterService clusterService
+        SDKClusterService clusterService,
+        ExtensionsRunner extensionsRunner
     ) {
         super(ADCancelTaskAction.NAME, actionFilters, taskManager);
         this.adTaskManager = adTaskManager;
         this.clusterService = clusterService;
+        this.extensionsRunner = extensionsRunner;
     }
 
     protected ADCancelTaskResponse newResponse(
         ADCancelTaskRequest request,
         List<ADCancelTaskNodeResponse> responses,
         List<FailedNodeException> failures
     ) {
-        return new ADCancelTaskResponse(clusterService.state().getClusterName(), responses, failures);
+        return new ADCancelTaskResponse(
+            new ClusterName(extensionsRunner.getEnvironmentSettings().get("cluster.name")),
```

Comment:
Instead of creating a new object every time. Can we have it in createComponent and pass along?

Reply:
I've explored this option but `createComponents` is invoked during `ExtensionsRunner` initialization, therefore we would be unable to retrieve the cluster name from the environment settings until after extension initialization since environment settings are set at that point.

Reply:
Can you check if we can store the cluster name as a global variable somewhere, the reason being cluster name would remain same?

Reply:
Sure, I think the best place to store this would be in the `SDKClusterService` itself. I'll add a cluster name variable to the `SDKClusterService` and set this during extension initialization as part of `updateSDKClusterService()`, which currently only updates the cluster settings. This way, we can just pull this information from the cluster service and avoid having to pass the extension runner to these transport actions.